xrft/padding.py

```
@@ -0,0 +1,218 @@
+        reflect_type,
+    )
+    # Pad the coordinates selected in pad_width
+    padded_coords = pad_coordinates(da.coords, pad_width)
+    # Assign the padded coordinates to the padded array
+    padded_da = padded_da.assign_coords(padded_coords)
+    # Return padded array
+    return padded_da
+
+
+def pad_coordinates(coords, pad_width):
+    """
+    Pad coordinates arrays according to the passed pad_width
+
+    Parameters
+    ----------
+    coords : dict-like object
+        Dictionary with coordinates as :class:`xarray.DataArray`.
+        Only the coordinates specified through ``pad_width`` will be padded.
+        Every coordinate that will be padded should be evenly spaced.
+    pad_width : dict-like object
+        Dictionary with the same keys as ``coords``. The coordinates specified
+        through ``pad_width`` are returned as padded.
+
+    Returns
+    -------
+    padded_coords : dict-like object
+        Dictionary with 1d-arrays corresponding to the padded coordinates.
+
+    Examples
+    --------
+
+    >>> import numpy as np
+    >>> import xarray as xr
+    >>> x = np.linspace(-4, -1, 4)
+    >>> y = np.linspace(-1, 4, 6)
+    >>> coords = {
+    ... "x": xr.DataArray(x, coords={"x": x}, dims=("x",)),
+    ... "y": xr.DataArray(y, coords={"y": y}, dims=("y",)),
+    ... }
+    >>> pad_width = {"x": 2}
+    >>> padded_coords = pad_coordinates(coords, pad_width)
+    >>> padded_coords["x"]
+    array([-6., -5., -4., -3., -2., -1.,  0.,  1.])
+    >>> padded_coords["y"]
+    <xarray.DataArray (y: 6)>
+    array([-1.,  0.,  1.,  2.,  3.,  4.])
+    Coordinates:
+      * y        (y) float64 -1.0 0.0 1.0 2.0 3.0 4.0
+
+
+    """
+    # Generate a dictionary with the original coordinates
+    padded_coords = {dim: coords[dim] for dim in coords}
+    # Start padding the coordinates that appear in pad_width
+    for dim in pad_width:
+        # Get the spacing of the selected coordinate
+        # (raises an error if not evenly spaced)
```

I am not sure if the correct behaviour is to error for unevenly spaced coordinates, maybe it is better to fall back on the existing behaviour of pad (particularly for applications in xarray outside of FTs)?

---

I agree that for a general case it should fall back to the default behaviour of the xarray `pad` method. Nevertheless, for `xrft` I think we should raise the error. If the coordinates are not evenly spaced, you won't be able to pass the padded grid to `fft` (or get awkward errors related to the presence of `NaN`s in the coordinates.

In summary:
- I would leave the error raising inside `xrft`
- I wouldn't add the error on an `xarray` implementation of this feature.